synapse/cortex.py

```diff
@@ -2081,6 +2105,194 @@ async def syncLayerNodeEdits(self, iden, offs, wait=True):
         async for item in layr.syncNodeEdits(offs, wait=wait):
             yield item
 
+    async def syncLayersEvents(self, offsdict=None, wait=True):
+        # syncLayersEvents - rename
+        '''
+        Yield (offs, layriden, STYP, item, meta) tuples for nodeedits for *all* layers, interspersed with add/del
+        layer messages.
+
+        STYP is one of the following constants:
+            SYNC_NODEEDITS:  item is a nodeedits (buid, form, edits)
+            SYNC_LAYR_ADD:   A layer was added (item and meta are empty)
+            SYNC_LAYR_DEL:   A layer was deleted (item and meta are empty)
+
+        Args:
+            offsdict(Optional(Dict[str,int])): starting nexus/editlog offset by layer iden.  Defaults to 0 for
+                unspecified layers or if offsdict is None.
+            wait(bool):  whether to pend and stream value until this layer is fini'd
+        '''
+        async def layrgenr(layr, startoff, endoff=None, newlayer=False):
+            if newlayer:
+                yield layr.addoffs, layr.iden, SYNC_LAYR_ADD, (), {}
+
+            wait = endoff is None
+
+            if not layr.isfini:
+
+                async for ioff, item, meta in layr.syncNodeEdits2(startoff, wait=wait):
+                    if endoff is not None and ioff >= endoff:  # pragma: no cover
+                        break
+
+                    yield ioff, layr.iden, SYNC_NODEEDITS, item, meta
+
+            if layr.isdeleted:
+                yield layr.deloffs, layr.iden, SYNC_LAYR_DEL, (), {}
+
+        # End of layrgenr
+
+        async for item in self._syncNodeEdits(offsdict, layrgenr, wait=wait):
+            yield item
+
+    async def syncIndexEvents(self, matchdef, offsdict=None, wait=True):
+        '''
+        Yield (offs, layriden, <STYPE>, <item>) tuples from the nodeedit logs of all layers starting
+        from the given nexus/layer offset (they are synchronized).  Only edits that match the filter in matchdef will
+        be yielded, plus EDIT_PROGRESS (see layer.syncIndexEvents) messages.
+
+        The format 4th element of the tuple depends on the STYP.  STYP is one of the following constants
+
+          SYNC_LAYR_ADD:  item is an empty tuple ()
+          SYNC_LAYR_DEL:  item is an empty tuple ()
+          SYNC_NODEEDITS: item is (buid, form, individual edit)) or (None, None, s_layer.EDIT_PROGRESS, (), ())
```

Comment:
singluar SYNC_NODEEDIT, also might be better to specify it is `(buid, form, etyp, vals, meta)` to have the correct length tuple, but `individual edit` does kinda explain it better